### PR TITLE
Support user id hashing

### PR DESCRIPTION
**Pull request by yossipapi**

### CODE REVIEW

File: plugins/search/providers/elastic_search/lib/model/items/ESearchUserItem.php [hunk 1]
```diff
@@ -39,7 +39,15 @@ class ESearchUserItem extends ESearchItem
...
-		ESearchUserFieldName::COUNTRY => array('ESearchItemType::EXACT_MATCH'=> ESearchItemType::EXACT_MATCH, 'ESearchItemType::STARTS_WITH'=> ESearchItemType::STARTS_WITH)
+		ESearchUserFieldName::COUNTRY => array('ESearchItemType::EXACT_MATCH'=> ESearchItemType::EXACT_MATCH, 'ESearchItemType::STARTS_WITH'=> ESearchItemType::STARTS_WITH),
+		ESearchUserFieldName::EXTERNAL_ID => array('ESearchItemType::EXACT_MATCH'=> ESearchItemType::EXACT_MATCH, 'ESearchItemType::PARTIAL'=> ESearchItemType::PARTIAL, 'ESearchItemType::STARTS_WITH'=> ESearchItemType::STARTS_WITH),
+		ESearchUserFieldName::IS_HASHED => array('ESearchItemType::EXACT_MATCH'=> ESearchItemType::EXACT_MATCH, 'ESearchItemType::EXISTS' => ESearchItemType::EXISTS),
+		ESearchUserFieldName::IS_ADMIN => array('ESearchItemType::EXACT_MATCH'=> ESearchItemType::EXACT_MATCH, 'ESearchItemType::EXISTS' => ESearchItemType::EXISTS),
+		ESearchUserFieldName::LOGIN_ENABLED => array('ESearchItemType::EXACT_MATCH'=> ESearchItemType::EXACT_MATCH, 'ESearchItemType::EXISTS' => ESearchItemType::EXISTS),
+	);
+	
+	protected static $booleanFields = array(
```
Comment (inbal-ben-david) [inline plugins/search/providers/elastic_search/lib/model/items/ESearchUserItem.php:49]: why only is hashed was added here?
is admin and login_enabled are also boolean

File: release-notes.md [hunk 2]
```diff
@@ -1,4 +1,43 @@
 # Rigel-18.19.0
+## Support user id hashing ##
+
+* Issue Type: Task
```
Comment (inbal-ben-david) [inline release-notes.md:4]: this got messed up, you have issue type and issue id twice

File: release-notes.md [hunk 3]
```diff
@@ -1,4 +1,43 @@
...
+* Issue ID: - Issue Type: Feature
+- Issue ID: PLAT-24004
+
+#### Deployment ####
+Execute the curl command, replace esearch_host, esearch_port and kaltura_kuser index (default it 'kaltura_kuser')
```
Comment (inbal-ben-david) [inline release-notes.md:9]: default is

File: release-notes.md [hunk 4]
```diff
@@ -1,4 +1,43 @@
...
+##### Note: command below is for elastic 7.x.x version, if you have different version, please refer to elastic documentations on how to update index mapping #####
+Elastic docs: https://www.elastic.co/guide/en/elasticsearch/reference/7.10/indices-put-mapping.html
+
+{ "external_id": { "type": "keyword", "normalizer": "kaltura_keyword_normalizer" }, "is_hashed": { "type": "boolean"} }
```
Comment (inbal-ben-david) [inline release-notes.md:13]: you put in the example external_id and is_hashed but on the curl you put the 4 new types
is line 13 even needed?

File: alpha/apps/kaltura/lib/myKuserUtils.class.php [hunk 5]
```diff
@@ -41,4 +41,21 @@ public static function sanitizeFields(array $values)
...
+			return $userId;
+		}
+		
+		$userId = self::normalizeKuserId($userId);
+		return hash_hmac("sha256", $userId, $encKey);
+	}
+	
+	public static function normalizeKuserId($userId)
+	{
+		return str_replace(' ', '', trim(strtolower($userId)));;
```
[the remark this answers is not in the document]
Comment (yossipapi) [reply alpha/apps/kaltura/lib/myKuserUtils.class.php:59]: Done

File: alpha/lib/model/BulkUploadResultKuser.php [hunk 6]
```diff
@@ -85,4 +86,7 @@ public function setTags($v)	{$this->putInCustomData(self::TAGS, $v);}
 	
 	public function getGroup()	{return $this->getFromCustomData(self::GROUP);}
 	public function setGroup($v)	{$this->putInCustomData(self::GROUP, $v);}
+	
+	public function getExternalId()	{return $this->getFromCustomData(self::GROUP);}
```
[the remark this answers is not in the document]
Comment (yossipapi) [reply alpha/lib/model/BulkUploadResultKuser.php:90]: Done

**Review by inbal-ben-david — APPROVED** [review]

**Pull request merged**